### PR TITLE
ACMS-1935: Add acquia/memcache-settings plugin.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #ACMS-1935

**Proposed changes**
Add acquia/memcache-settings plugin in DRS.

**Alternatives considered**
NA

**Testing steps**
Follow from ticket.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
         "php": ">=7.4",
         "composer-plugin-api": "^2",
         "acquia/drupal-environment-detector": "^1.5.3",
+        "acquia/memcache-settings": "^1.2",
```
Comment (vishalkhode1) [inline composer.json:21]: @rajeshreeputra This is not required. Let's keep it, similar to acquia/blt.

File: composer.json [hunk 2]
```diff
@@ -51,6 +52,12 @@
             "Acquia\\Drupal\\RecommendedSettings\\": "src/"
         }
     },
+    "repositories": {
```
Comment (vishalkhode1) [inline composer.json:55]: @rajeshreeputra Looks like this is also not required.

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (rajeshreeputra) [reply composer.json:21]: sure, as discussed, removed.

File: composer.json (hunk 2 above)
Comment (rajeshreeputra) [reply composer.json:55]: sure, as discussed, removed.

File: settings/default.includes.settings.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+$additionalSettingsFiles = [
+  // e.g,( DRUPAL_ROOT . "/sites/$site_dir/settings/foo.settings.php" )
+];
+
+foreach ($additionalSettingsFiles as $settingsFile) {
+  if (file_exists($settingsFile)) {
+    // phpcs:ignore
+    require $settingsFile;
+  }
+}
```
Comment (chandan-singh7929) [inline settings/default.includes.settings.php:51]: Why this file is required?

File: src/Settings.php [hunk 4]
```diff
@@ -149,6 +149,10 @@ public function getSettings(): array {
     $settings['default_glob_settings_file'] = $this->getWebRootPath . "/sites/settings/default.global.settings.php";
     $settings['global_settings_file'] = $this->getWebRootPath . "/sites/settings/global.settings.php";
 
+    // Generate default.includes.settings.php.
+    $settings['recommended_includes_settings_file'] = $this->getSettingsPackagePath() . '/settings/default.includes.settings.php';
+    $settings['default_includes_settings_file'] = $this->getWebRootPath . "/sites/settings/default.includes.settings.php";
+
```
Comment (chandan-singh7929) [inline src/Settings.php:155]: I don't think this changes is need as part of this ticket.

File: src/Settings.php [hunk 5]
```diff
@@ -161,6 +165,7 @@ public function generateSettings(): void {
     $copy_map = [
       $setting_files['recommended_local_settings_file'] => $setting_files['default_local_settings_file'],
       $setting_files['default_local_settings_file'] => $setting_files['project_local_settings_file'],
+      $setting_files['recommended_includes_settings_file'] => $setting_files['default_includes_settings_file'],
```
Comment (chandan-singh7929) [inline src/Settings.php:168]: This one as well

**Review by chandan-singh7929 — COMMENTED** [review]
@rajeshreeputra - Please take a look

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**